### PR TITLE
DEM-875 Column selection component

### PR DESCRIPTION
**Pull request by yordanalexandrov**
+ Implemented column selection component using a mat-select
+ Input is of type SelectionColumnGroup
+ Output is an array of SelectionColumn
+ Added cypress tests

### CODE REVIEW

File: cypress/test/search-column-selector.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    step.getSelectedColumns().should('contain', 'Username');
+    // When I press the select
+    step.getColumnSelectDropdown().click();
+    // I should see column groups
+    step.getColumnGroups().should('be.visible');
+    // And they should contain 'Default Columns' and 'Other Columns'
+    step.getColumnGroups().should('contain', 'Default Columns').and('contain', 'Other Columns');
+    // When I look at 'Default Columns Group' I should see columns 'Username' and 'Name'
+    step.getColumnGroups().filter('[appCypressData=column-group-0]')
+        .find('mat-option').should('contain', 'Username').and('contain', 'Name');
```
Comment (svilenvelikov) [inline cypress/test/search-column-selector.spec.ts]: You can create a method in the steps for obtaining concrete option by index maybe

File: src/app/search-column-selector/search-column-selector.component.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="toolbar">
+  <button (click)="changeSelection()" appCypressData="change-selection-button">Change selection
+  </button>
+  <onto-search-column-selector [defaultColumns]="defaultColumns"
+                               (selectionChanged)="onColumnSelection($event)"
+                               [columnGroups]="columnGroups"
+                               [selectedColumns]="outsideSelection"></onto-search-column-selector>
+  <div *ngFor="let column of selectedColumns" appCypressData="selected-columns">{{column.label}}
+  </div>
+</div>
```
Comment (svilenvelikov) [inline src/app/search-column-selector/search-column-selector.component.html:10]: Remove the empty search-column-selector.component.scss file below

File: projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
+.column-selector-component {
```
Comment (svilenvelikov) [inline projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.scss]: Is this selector correct? I couldn't find where it's used. Did you mean `onto-column-selector-component`?

File: projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    box-sizing: border-box;
+    max-width: 100%;
+  }
+
+  .form-field {
+    display: inherit;
+  }
+}
+
+button.column-reset-button {
```
Comment (svilenvelikov) [inline projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.scss]: Move this selector inside `column-selector-component` and also remove the button tag selector.

File: projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  defaultColumns: SelectionColumn[];
+
+  /**
+   * Fired when the dropdown panel is closed.
+   * Sends an array {@link SelectionColumn}
+   */
+  @Output()
+  selectionChanged: EventEmitter<SelectionColumn[]> = new EventEmitter<SelectionColumn[]>();
+
+  constructor() {
```
Comment (svilenvelikov) [inline projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.ts]: Remove this empty constructor

File: projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    const defaultColumns = changes?.defaultColumns?.currentValue;
+    const selectedColumns = changes?.selectedColumns?.currentValue;
+    if (defaultColumns) {
```
Comment (svilenvelikov) [inline projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.ts]: Don't you need to check if there are some default values actually? As the defaultColumns is an array, if you check for the reference only, you'll get true even if it's empty

File: projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.scss (hunk 3 above)
Comment (yordanalexandrov) [reply projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.scss]: yes i ment `onto-column-selector-component`

**Review by svilenvelikov — APPROVED** [review]

**Review by teodossidossev — APPROVED** [review]

**Pull request merged**